src/action-lib/appshell.c

```diff
@@ -0,0 +1,59 @@
+/* 
```

Comment:
I still don't exactly understand how an app has it's own shell. Can you explain?